### PR TITLE
Add dig.cname function

### LINKED ISSUE #61991: [FEATURE REQUEST] add CNAME support to the dig module
**Describe the solution you'd like**
Support returning CNAME records via dig.

**Describe alternatives you've considered**
N/A

**Additional context**
Other record types are supported via `cmd.run_all` invocations of the `dig` command.

**Please Note**
If this feature request would be considered a substantial change or addition, this should go through a SEP process here https://github.com/saltstack/salt-enhancement-proposals, instead of a feature request.

### PR DESCRIPTION
**Pull request by nicholasmhughes**
### What does this PR do?
This PR adds a function to the `dig` execution module which supports returning CNAME records for a given host.

### What issues does this PR fix or reference?
Fixes: #61991

### New Behavior
New function to support CNAME record queries via dig.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by nicholasmhughes** [comment]
re-run pr-ubuntu-2004-amd64-py3-pycryptodome-pytest

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: salt/modules/dig.py [hunk 1]
```diff
@@ -129,6 +129,33 @@ def AAAA(host, nameserver=None):
...
+    cmd = __salt__["cmd.run_all"](dig, python_shell=False)
+    # In this case, 0 is not the same as False
+    if cmd["retcode"] != 0:
+        log.warning(
+            "dig returned exit code '%s'.",
+            cmd["retcode"],
+        )
+        return []
+
+    return cmd["stdout"]
```
[the remark this answers is not in the document]
Comment (nicholasmhughes) [reply salt/modules/dig.py:156]: The RFC only allows for a one-to-one relationship for CNAMEs.

**Comment by nicholasmhughes** [comment]
re-run pr-alma-8-x86_64-py3-pytest

**Comment by nicholasmhughes** [comment]
re-run pr-centos-7-x86_64-py3-pytest

**Comment by nicholasmhughes** [comment]
re-run pr-centos-7-x86_64-py3-pytest

**Comment by nicholasmhughes** [comment]
re-run pr-centosstream-8-x86_64-py3-pytest

**Comment by nicholasmhughes** [comment]
re-run pr-debian-9-amd64-py3-pytest

**Comment by nicholasmhughes** [comment]
re-run pr-debian-9-amd64-py3-pytest

**Pull request merged**